src/GovUk.Education.ExploreEducationStatistics.Data.Model/Migrations/Current_Routine_RebuildIndexes.sql

```diff
@@ -1,120 +1,298 @@
+                   JOIN sys.schemas sch ON sch.schema_id = obj.schema_id
+          WHERE ips.alloc_unit_type_desc = 'IN_ROW_DATA') AS Fragmented;
+
+    ALTER TABLE #Stats
+        ADD PRIMARY KEY (Id);
+
+    /*-----------------------------------------------------------------------*/
+
+    -- Create result tables if necessary
+    IF OBJECT_ID(N'__Log_RebuildIndexes', N'U') IS NULL
+    CREATE TABLE __Log_RebuildIndexes
+    (
+        Id         INT IDENTITY (1,1),
+        StartTime  DATETIME2 NOT NULL,
+        EndTime    DATETIME2,
+        HitTimeout BIT,
+        ReportSent DATETIME2,
+        PRIMARY KEY (Id),
+    );
+
+    IF OBJECT_ID(N'__Log_RebuildIndexesAlterIndexes', N'U') IS NULL
+    CREATE TABLE __Log_RebuildIndexesAlterIndexes
+    (
+        Id               INT IDENTITY (1,1),
+        RunId            INT           NOT NULL,
+        IndexName        NVARCHAR(MAX) NOT NULL,
+        SchemaName       NVARCHAR(MAX) NOT NULL,
+        ObjectName       NVARCHAR(MAX) NOT NULL,
+        StartTime        DATETIME2,
+        EndTime          DATETIME2,
+        StartFragPercent FLOAT,
+        EndFragPercent   FLOAT,
+        ActionRequired   NVARCHAR(MAX),
+        HitTimeout       BIT DEFAULT 0,
+        PRIMARY KEY (Id),
+        FOREIGN KEY (RunId) REFERENCES __Log_RebuildIndexes (Id),
+    );
 
+    -- With this table:
+    -- - if StartTime and EndTime are set, we ran UPDATE STATISTICS on the table
+    -- - if StartTime and EndTime aren't set, that means we didn't process the table due exceeding @StopTime
+    -- - if only StartTime is set, then no indexes related to the table were processed so we skipped the UPDATE STATISTICS call
+    IF OBJECT_ID(N'__Log_RebuildIndexesUpdateStatistics', N'U') IS NULL
+    CREATE TABLE __Log_RebuildIndexesUpdateStatistics
+    (
+        Id        INT IDENTITY (1,1),
+        RunId     INT           NOT NULL,
+        TableName NVARCHAR(MAX) NOT NULL,
+        StartTime DATETIME2,
+        EndTime   DATETIME2,
+        PRIMARY KEY (Id),
+        FOREIGN KEY (RunId) REFERENCES __Log_RebuildIndexes (Id),
+    );
+
+    -- Create new row for this stored procedure run
+    INSERT INTO __Log_RebuildIndexes (StartTime)
+    VALUES (GETUTCDATE())
+    -- The Id is generated, but we want it, so set @RunId equal to it here
+    SET @RunId = SCOPE_IDENTITY();
+
+    -- Create rows for indexes
+    SELECT @StatsCount = COUNT(Id) FROM #Stats;
+
+    SET @StatsRowIndex = 1;
+
+    INSERT INTO __Log_RebuildIndexesAlterIndexes (RunId, IndexName, SchemaName,
+                                                  ObjectName, StartFragPercent, ActionRequired)
+    SELECT @RunId,
+           IndexName,
+           SchemaName,
+           ObjectName,
+           FragPercent,
+           ActionRequired
+    FROM #Stats;
+
+    -- Create rows for tables
+    INSERT INTO __Log_RebuildIndexesUpdateStatistics (RunId, TableName)
+    SELECT @RunId,
+           ObjectName
+    FROM #TableList TL;
```

Comment:
In the latest set of changes `tl.ObjectName` was changed to `ObjectName` following a previous comment,

but rather than

```sql
INSERT INTO __Log_RebuildIndexesUpdateStatistics (RunId, TableName)
    SELECT @RunId,
           ObjectName
    FROM #TableList TL;
```

this can be

```sql
INSERT INTO __Log_RebuildIndexesUpdateStatistics (RunId, TableName)
    SELECT @RunId,
           ObjectName
    FROM #TableList;
```